lib/Provider/MailProvider.php

```diff
@@ -0,0 +1,238 @@
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @return string				label/name of this provider (e.g. Plain Old IMAP/SMTP)
+	 */
+	public function label(): string {
+
+		return 'Mail Application';
+
+	}
+
+	/**
+	 * determain if any services are configured for a specific user
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @param string $userId		user id
+	 *
+	 * @return bool 				true if any services are configure for the user
+	 */
+	public function hasServices(string $userId): bool {
+
+		return (count($this->listServices($userId)) > 0);
+
+	}
+
+	/**
+	 * retrieve collection of services for a specific user
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @param string $userId			user id
+	 *
+	 * @return array<string,IService>	collection of service id and object ['1' => IServiceObject]
+	 */
+	public function listServices(string $userId): array {
+
+		try {
+			// retrieve service(s) details from data store
+			$accounts = $this->AccountService->findByUserId($userId);
+		} catch (\Throwable $th) {
+			return [];
+		}
+		// construct temporary collection
+		$services = [];
+		// add services to collection
+		foreach ($accounts as $entry) {
+			// extract values
+			$serviceId = (string) $entry->getId();
+			$label = $entry->getName();
+			$address = new MailAddress($entry->getEmail(), $entry->getName());
+			$identity = new MailServiceIdentity();
+			$location = new MailServiceLocation();
+			// add service to collection
+			$services[$serviceId] = new MailService($this->container, $userId, $serviceId, $label, $address, $identity, $location);
+		}
+		// return list of services for user
+		return $services;
+
+	}
+
+	/**
+	 * retrieve a service with a specific id
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @param string $userId			user id
+	 * @param string $serviceId			service id
+	 *
+	 * @return IService|null			returns service object or null if none found
+	 */
+	public function findServiceById(string $userId, string $serviceId): IService | null {
+
+		// evaluate if id is a number
+		if (is_numeric($serviceId)) {
+			try {
+				// retrieve service details from data store
+				$account = $this->AccountService->find($userId, (int) $serviceId);
+			} catch(\Throwable $th) {
+				return null;
+			}
+		}
+		// evaliate if service details where found
+		if ($account instanceof Account) {
+			// extract values
+			$serviceId = (string) $account->getId();
+			$label = $account->getName();
+			$address = new MailAddress($account->getEmail(), $account->getName());
+			$identity = new MailServiceIdentity();
+			$location = new MailServiceLocation();
+			// return mail service object
+			return (new MailService($this->container, $userId, $serviceId, $label, $address, $identity, $location));
+		}
+
+		return null;
+		
+	}
+
+	/**
+	 * retrieve a service for a specific mail address
+	 *
+	 * @since 2024.05.25
+	 *
+	 * @param string $userId			user id
+	 * @param string $address			mail address (e.g. test@example.com)
+	 *
+	 * @return IService					returns service object or null if none found
```

Comment:
```suggestion
	 * @return IService
```

Self documenting code really